### ISSUE TITLE
[Proton-TKG] Name Addons Don't Apply With Makepkg

### ISSUE THREAD

**Issue by Orangian** [Frogging-Family/wine-tkg-git#688]
Hi, when building proton-tkg with a custom name addon through makepkg, the package name still remains proton-tkg-git. The file paths have the name addon in it, though.

Addon here is qern, this snippet from the log shows `usr/share/steam/compatibilitytools.d/proton_tkg_makepkg.qern/files/lib64/libgstvideo-1.0.so` 

The output file here, however, is `proton-tkg-git-7.1.r4.gecf05ee1-1-x86_64.pkg.tar.zst`

**Comment by Orangian** [comment]
It appears that in the PKGBUILD, the package name is set to proton-tkg-git, with no variables to account for _PROTON_NAME_ADDON being set to anything different in advanced-customization.cfg, so no matter what, any output with the PKGBUILD will be named proton-tkg-git. I don't really know how to fix this, so I can't make a pull request at the moment, but I'll try to make one later.

**Comment by Tk-Glitch** [comment]
Due to the way packaged proton works, you can't have multiple installed at a time by design. The preferred and recommended way is to use regular, non-packaged proton-tkg if you want to install multiple builds at a time and use all features. The package version is a huge hack bypassing a large amount of basic pacman rules :frog:

**Comment by ryanmusante** [comment]
@Tk-Glitch does that include when one version of proton is built from makepkg and one version is installed from AUR?

**Comment by Orangian** [comment]
> Due to the way packaged proton works, you can't have multiple installed at a time by design. The preferred and recommended way is to use regular, non-packaged proton-tkg if you want to install multiple builds at a time and use all features. The package version is a huge hack bypassing a large amount of basic pacman rules 🐸

Ah, I see. I actually wasn't trying to install multiple at once, I just build a few different versions for different systems on one of my servers. I do see why you designed it this way, so I guess I'll edit the pkgbuild during the build to change the package name. Thanks for making all these scripts!

**Comment by Tk-Glitch** [comment]
> @Tk-Glitch does that include when one version of proton is built from makepkg and one version is installed from AUR?

If you have different internal names for each package with explicitly different paths, that should be okay. If not, you might break your Steam client.

The initial huge issue was that the Steam client didn't handle compat tools removal when bound to a game. So you uninstalled a proton package that was in use by a game or multiple and you ended up with those games getting nuked by Steam and other issues. I made the uninstaller thing as a workaround so we could wait for a Steam-side solution. That huge stopper was finally fixed at some point so we *could* rework it to not be as limiting, but ultimately packaging proton has other limitations we'd need to add more hacks for in the long run, and that's not too great.
The current implementation answers the initial request for a way to make proton-tkg pacman packages -with known limitations- but it's kinda robust for what it does and shouldn't break that often, even though it's clearly not the recommended way to use proton, be it -tkg or not.
